### PR TITLE
feat(kubernetes): Projected Volume Replacement

### PR DESCRIPTION
**Pull request by kskewes**
Closes https://github.com/spinnaker/spinnaker/issues/4481 https://github.com/spinnaker/spinnaker/issues/6002

- add Groovy tests for artifact replacement (not sure if this file can be deleted as duped in Java?)
- add Java tests for artifact replacement of Projected Volume ConfigMap's and Secrets
- add Replacer functions
- use Replacer functions in Kubernetes kind handlers.

### CODE REVIEW

File: clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/ReplacerTest.java [hunk 1]
```diff
@@ -28,35 +28,7 @@
...
-import io.kubernetes.client.openapi.models.V1ReplicaSetBuilder;
-import io.kubernetes.client.openapi.models.V1SecretEnvSource;
-import io.kubernetes.client.openapi.models.V1SecretEnvSourceBuilder;
-import io.kubernetes.client.openapi.models.V1SecretKeySelector;
-import io.kubernetes.client.openapi.models.V1SecretKeySelectorBuilder;
-import io.kubernetes.client.openapi.models.V1SecretVolumeSource;
-import io.kubernetes.client.openapi.models.V1SecretVolumeSourceBuilder;
-import io.kubernetes.client.openapi.models.V1Volume;
-import io.kubernetes.client.openapi.models.V1VolumeBuilder;
+import io.kubernetes.client.openapi.models.*;
```
Comment (kskewes) [inline clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/ReplacerTest.java:31]: Ah, IntelliJ did this on its own accord, happy to revert to specifics.

**Comment by kskewes** [comment]
Validated this in Minnaker (very handy btw) with [pipeline gist](https://gist.github.com/kskewes/3460ff360939dbe3fabb4ece841712ab)

File: clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/ReplacerTest.java (hunk 1 above)
Comment (ezimanyi) [reply clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/ReplacerTest.java:31]: No worries; in general I think we prefer specific imports, but as our code formatter doesn't enforce it I don't usually comment either way on PRs.  (And in this case it is replacing *a lot* of imports.)

**Review by ezimanyi — APPROVED** [review]
This looks great, thanks for this change (and for the very thorough tests)!

**Pull request merged**